addon/components/validated-button.js

```diff
@@ -8,7 +9,11 @@ export default Component.extend({
 
   tagName: 'button',
 
-  classNameBindings: ['class'],
+  classNameBindings: ['class', 'loadingClass'],
```

Comment:
Is `class` needed after all?

Reply:
Yes. We want `loading` to be a boolean. Since we cannot do a generic class mapping, we need to solve this with a computed property